lib/opte/src/engine/predicate.rs

```diff
@@ -505,12 +506,53 @@ impl Predicate {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub enum Match<T> {
+    Exact(T),
+    Range(RangeInclusive<T>),
+}
+
+impl<T> Match<T>
+where
+    T: PartialEq + PartialOrd,
+{
+    pub(crate) fn is_match(&self, val: &T) -> bool {
+        match self {
+            Self::Exact(target) => val == target,
+            Self::Range(range) => range.contains(val),
+        }
+    }
+}
+
+impl<T: Display> Display for Match<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Exact(target) => write!(f, "={target}"),
+            Self::Range(range) => {
+                write!(f, "∈({}..={})", range.start(), range.end())
```

Comment:
Debug formatter on `RangeInclusive` will also give you the `{}..={}` output.

Reply:
Would that not provide `{:?}..={:?}`? Although it would probably be less space consuming than having long display formats like "router solicitation" constantly appearing in the rule set.

Reply:
ah right, this is `Display` impl not `Debug`.